src/Core/Transmission.EventHub/EventHubTransmitter.cs

```diff
@@ -105,20 +89,13 @@ public void Dispose()
             {
                 _disposeToken.Cancel();
 
-                if (!_sendJob.Stopped && !_storeJob.Stopped)
-                {
-                    WaitHandle.WaitAll(new[]
-                    {
-                        _sendJob.WaitHandle,
-                        _storeJob.WaitHandle
-                    }, TimeSpan.FromSeconds(5));
-                }
+                SpinWait.SpinUntil(() =>
```

Comment:
Spin until is not good when in an async context. This is basically blocking this thread. Task.WhenAll was the better choice here.

Reply:
sorry did not see that this is in dispose :) But still would use Task.WaitAll

Reply:
It's in dispose